upgrade/helpers/tasks.py

```diff
@@ -801,67 +801,111 @@ def check_ntpd():
     """
-    env.disable_known_hosts = True
-    # setting up foreman-maintain repo
-    setup_foreman_maintain_repo()
     if settings.upgrade.distribution != 'cdn':
-        for repo, repodata in CUSTOM_SAT_REPO.items():
-            if repo != 'maintenance':
-                repository_setup(**repodata)
+        repository_setup(**CUSTOM_SAT_REPO['satellite'])
 
 
-def setup_foreman_maintain_repo():
+def setup_maintenance_repo():
     """
-    Task which setup repo for foreman-maintain.
+    Task which setups maintenance repo.
     """
-    # setting up foreman-maintain repo
     if settings.upgrade.distribution == 'cdn':
         enable_repos(RH_CONTENT['maintenance']['label'])
     else:
         repository_setup(**CUSTOM_SAT_REPO['maintenance'])
 
 
-def hammer_config():
+def setup_capsule_repo(fetch_content_from_sat=True):
     """
-    Use to update the hammer config file on the satellite
+    Task which setups capsule repo.
     """
-    run('mkdir -p /root/.hammer/cli.modules.d')
-    hammer_file = StringIO()
-    hammer_file.write('--- \n')
-    hammer_file.write(' :foreman: \n')
-    hammer_file.write('  :username: admin\n')
-    hammer_file.write('  :password: changeme \n')
-    put(local_path=hammer_file,
-        remote_path='/root/.hammer/cli.modules.d/foreman.yml')
-    hammer_file.close()
+    if settings.upgrade.distribution != 'cdn':
+        if fetch_content_from_sat:
+            product_label = CUSTOM_CONTENT['capsule']['prod']
+            repo_label = CUSTOM_CONTENT['capsule']['reposet']
+            enable_repos(f'Default_Organization_{product_label}_{repo_label}')
+        else:
+            repository_setup(**CUSTOM_SAT_REPO['capsule'])
 
 
-def upgrade_using_foreman_maintain(satellite=True):
+def setup_capsule_maintenance_repo(fetch_content_from_sat=True):
+    """
+    Task which setups maintenance repo on capsule.
+    """
+    if settings.upgrade.distribution == 'cdn':
+        enable_disable_repo(
+            enable_repos_name=RH_CONTENT['maintenance']['label'],
+            disable_repos_name=RH_CONTENT['capsule']['label'],
+        )
+    elif fetch_content_from_sat:
+        maintenance_product = CUSTOM_CONTENT['maintenance']['prod']
+        maintenance_repo = CUSTOM_CONTENT['maintenance']['reposet']
+        capsule_product = CUSTOM_CONTENT['capsule']['prod']
+        capsule_repo = CUSTOM_CONTENT['capsule']['reposet']
+        enable_disable_repo(
+            enable_repos_name=f'Default_Organization_{maintenance_product}_{maintenance_repo}',
+            disable_repos_name=f'Default_Organization_{capsule_product}_{capsule_repo}',
+        )
+    else:
+        repository_setup(**CUSTOM_SAT_REPO['maintenance'])
+
+
+def foreman_maintain_self_upgrade(zstream=False, fetch_content_from_sat=False):
+    """
+    Install the latest fm rubygem-foreman_maintain to get the latest y-stream upgrade path.
+    """
+    if zstream:
+        run('foreman-maintain upgrade list-versions', warn_only=True)
```

Comment:
Isn't it possible to have scenarios where we can receive updates in the maintenance package for the Z-stream?

Reply:
I believe this command will check if any updates are available and, if so, will update them.